src/lib/addons/datadog.ts

```diff
@@ -26,12 +30,18 @@ interface DDRequestBody {
 export default class DatadogAddon extends Addon {
     private msgFormatter: FeatureEventFormatter;
 
-    constructor(config: IAddonConfig) {
+    private flagResolver: IFlagResolver;
+
+    constructor(
+        config: IAddonConfig,
+        { flagResolver }: Pick<IUnleashConfig, 'flagResolver'>,
```

Comment:
Maybe you have a reason but this feels a bit convoluted. In order to better express the DatadogAddon dependencies I think I would extend `IAddonConfig` to include a flag resolver.

```ts
interface IDatadogAddonConfig extends IAddonConfig {
  flagResolver: IFlagResolver;
}

// ...

this.flagResolver = config.flagResolver
```

Alternatively, simply accept an IFlagResolver here:

```ts
constructor(config: IAddonConfig, flagResolver: IFlagResolver)
```

Reply:
Went with your first suggestion, thanks!